designer/server/src/test/scala/pl/touk/nussknacker/ui/api/MigrationApiHttpServiceBusinessSpec.scala

```diff
@@ -0,0 +1,142 @@
+package pl.touk.nussknacker.ui.api
+
+import io.circe.syntax._
+import io.restassured.RestAssured.`given`
+import io.restassured.module.scala.RestAssuredSupport.AddThenToResponse
+import org.scalatest.freespec.AnyFreeSpecLike
+import pl.touk.nussknacker.engine.api.graph.ScenarioGraph
+import pl.touk.nussknacker.engine.api.process.ProcessName
+import pl.touk.nussknacker.engine.build.ScenarioBuilder
+import pl.touk.nussknacker.restmodel.validation.ValidationResults.{
+  NodeValidationError,
+  NodeValidationErrorType,
+  ValidationResult
+}
+import pl.touk.nussknacker.test.base.it.{NuItTest, WithRichConfigScenarioHelper}
+import pl.touk.nussknacker.test.config.WithRichDesignerConfig.TestCategory.Category1
+import pl.touk.nussknacker.test.config.{WithMockableDeploymentManager, WithRichDesignerConfig}
+import pl.touk.nussknacker.test.processes.WithRichScenarioActivitySpecAsserts
+import pl.touk.nussknacker.test.{NuRestAssureExtensions, NuRestAssureMatchers, RestAssuredVerboseLogging}
+import pl.touk.nussknacker.ui.process.marshall.CanonicalProcessConverter
+
+class MigrationApiHttpServiceBusinessSpec
+    extends AnyFreeSpecLike
+    with NuItTest
+    with WithRichDesignerConfig
+    with WithRichConfigScenarioHelper
+    with WithMockableDeploymentManager
+    with WithRichScenarioActivitySpecAsserts
+    with NuRestAssureExtensions
+    with NuRestAssureMatchers
+    with RestAssuredVerboseLogging {
+
+  "The endpoint for scenario migration between environments should" - {
+    "migrate scenario and add update comment" in {
+      given()
+        .applicationState(
+          createSavedScenario(exampleScenario, Category1)
+        )
+        .when()
+        .basicAuthAllPermUser()
+        .jsonBody(validRequestData)
+        .post(s"$nuDesignerHttpAddress/api/migrate")
+        .Then()
+        .statusCode(200)
+        .verifyCommentExists(exampleProcessName.value, "Scenario migrated from DEV by allpermuser")
```

Comment:
a comment prove that migration is done? :) I thought that it is a minor side effect and new shape of scenario graph is the most important :)

Reply:
You're right I should request for that new shape.